hack/generator/pkg/astmodel/object_type.go

```diff
@@ -183,6 +183,13 @@ func (objectType *ObjectType) AsType(codeGenerationContext *CodeGenerationContex
 	}
 }
 
+// AsZero renders an expression for the "zero" value of the type
+// types allows TypeName to resolve to the underlying type
+// ctx allows current imports to be correctly identified where needed
+func (objectType *ObjectType) AsZero(types Types, ctx *CodeGenerationContext) dst.Expr {
+	panic("cannot create a zero value for an object type")
```

Comment:
Why is this the case? Shouldn't the zero value of `SomeType` be `SomeType{}`? Basically it should be what you'd get by doing `var thing SomeType`.

Reply:
I think I might understand this now.

Reply:
The problem is that an ObjectType has no name - so we can't generate `SomeType{}`. Where the type has a name, the zero value is generated by `TypeName`. I've added a comment to make this more explicit.